modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java

```diff
@@ -684,7 +684,7 @@ public void pushTemperatureData(final DeviceJSON dataMsg, @Context HttpServletRe
 			response.setStatus(Response.Status.CONFLICT.getStatusCode());
 			return;
 		}
-
+        DataHolder.getThisInstance().setTemperature(deviceId, temperature, Calendar.getInstance().getTimeInMillis());
 		if (!publishToDAS(dataMsg.owner, dataMsg.deviceId, dataMsg.value)) {
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```

Comment:
put appropriate error log messages as a way of conveying an error has occurred at server-side to system administrators.